solutions/cses-1073.mdx

```diff
@@ -108,6 +108,39 @@ public class Towers {
 ```
 </JavaSection>
 
+<PySection>
+
+```python
+input()
+# Store the topmost cube of each tower
+# Towers are in non-decreasing order for sure
+towers = []
+for cube in map(int, input().split()):
+	# Binary search
+	left = 0
+	right = len(towers) - 1
+	tower_idx = -1
+	while left <= right:
+		mid = (left + right) // 2
+		if towers[mid] <= cube:
+			left = mid + 1
+		else:
+			right = mid - 1
+			tower_idx = mid
+
+	# If there exists a satisfying tower, add the cube to that tower
+	# and update the top element of the tower
+	if tower_idx != -1:
+		towers[tower_idx] = cube
+	# If there aren't any suitable towers, append new tower to end of
+	# array
```

Comment:
and yeah, no need for the 2 lines here(?)

Reply:
(line breaks are done manually by feeling lol)

Reply:
oh what text editor you using?

Reply:
~~notepad in most of the time~~
I have sublime text as well (ik there is a plugin that do the stuff)